shared/fs/row/rows-container.js

```diff
@@ -177,14 +178,20 @@ const mapStateToProps = state => ({
   _username: state.config.username,
 })
 
-const mapDispatchToProps = dispatch => ({})
+const mapDispatchToProps = dispatch => ({
+  onAttach: (parentPath: Types.Path, paths: Array<string>) => {
+    paths.forEach(localPath => dispatch(FsGen.createUpload({localPath, parentPath})))
+  },
+})
 
 // $FlowIssue
 const mergeProps = (s, d, o: OwnProps) => ({
   destinationPickerIndex: o.destinationPickerIndex,
   items: getItemsFromStateProps(s, o.path, o.sortSetting),
+  onAttach: d.onAttach,
   path: o.path,
   routePath: o.routePath,
+  writable: s._pathItems.get(o.path, Constants.unknownPathItem).writable,
```

Comment:
Instead of adding stuff here, can we instead make `DropTarget` itself a connected component? This component is also used in destination-picker, and it'd get complicated if we start adding these in.

Reply:
Avoided writable by setting onAttach only if it is writable. Thus no extra props.

Reply:
It'd still get used by destination-picker though. I think making `DropTarget` connected is cleaner, because 1) The Rows component should only deal with rows, but not stuff like drag and drop; 2) The Rows component is used in the destination picker too; having it in Rows would cause the destination picker to be drag/drop-able as well, which I assume we don't want; 3) if we ever want this to be used somewhere else, we a decoupled `DropTarget` that's connected by itself will be easier to use.